app/src/org/commcare/dalvik/activities/HomeActivityUIController.java

```diff
@@ -0,0 +1,404 @@
+
+    public HomeActivityUIController(CommCareHomeActivity activity) {
+        this.activity = activity;
+    }
+
+    public View getTopBanner() {
+        return mTopBanner;
+    }
+
+    public void setupUI() {
+        setMainView();
+        adapter = new HomeScreenAdapter(activity);
+        mTopBanner = View.inflate(activity, R.layout.grid_header_top_banner, null);
+        setupGridView();
+    }
+
+    private void setMainView() {
+        // TODO: discover why Android is not loading the correct layout from layout[-land]-v10 and remove this
+        if (usingNewLayout()) {
+            activity.setContentView(R.layout.mainnew_modern_v10);
+        } else {
+            activity.setContentView(R.layout.mainnew_modern);
+        }
+    }
+
+    private boolean usingNewLayout() {
+        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.GINGERBREAD_MR1;
+    }
+
+    private void setupGridView() {
+        final View grid = activity.findViewById(R.id.home_gridview_buttons);
+
+        if (usingNewLayout()) {
+            newGridView = (StaggeredGridView) grid;
+            newGridView.addHeaderView(mTopBanner);
+            newGridView.setAdapter(adapter);
+        } else {
+            gridView = (GridViewWithHeaderAndFooter)grid;
+            gridView.addHeaderView(mTopBanner);
+            gridView.setAdapter(adapter);
+        }
+
+        grid.getViewTreeObserver().addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
+            @SuppressLint("NewApi")
+            @Override
+            public void onGlobalLayout() {
+                if (Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN) {
+                    grid.getViewTreeObserver().removeGlobalOnLayoutListener(this);
+                } else {
+                    grid.getViewTreeObserver().removeOnGlobalLayoutListener(this);
+                }
+                grid.requestLayout();
+                adapter.notifyDataSetChanged(); // is going to populate the grid with buttons from the adapter (hardcoded there)
+                configUI();
+            }
+        });
+    }
+
+    protected void configUI() {
+        if (CommCareApplication._().getCurrentApp() == null) {
+            // This method depends on there being a seated app, so don't proceed with it if we
+            // don't have one
+            return;
+        }
+        setupVersion();
+        setupButtons();
+        activity.rebuildMenus();
+    }
+
+    private void setupVersion() {
+        TextView version = (TextView) activity.findViewById(R.id.str_version);
+        if (version != null) {
+            version.setText(CommCareApplication._().getCurrentVersionString());
+        }
+    }
+
+    private void setupButtons() {
+        refreshDataFromSyncDetails();
+        setupStartButton();
+        setupIncompleteFormsButton();
+        setupLogoutButton();
+        setupViewOldFormsButton();
+        setupSyncButton();
+    }
+
+    // region - setup methods for all buttons
```

Comment:
Can we take out `region` comments wherever we see them? I'm not a fan. Good method naming and organization should do that job.